src/views/MediaChannels/components/GlobalSearch.tsx

```diff
@@ -0,0 +1,170 @@
+import { Chip, Grid, TextField, Typography } from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import throttle from "lodash/throttle";
+import * as React from "react";
+import { FC, useMemo } from "react";
+import { useIntl } from "react-intl";
+
+import {
+  fetchOptions,
+  Option,
+  OptionType,
+} from "@temp/core/apiLayer/productSearchOptionsService";
+import { SearchProductCriteria } from "@temp/core/apiLayer/productService";
+
+import { messages } from "../messages";
+
+const normalizeOptions = (options: Option[]) =>
+  options.reverse().reduce(
+    (acc, option) => {
+      if (
```

Comment:
not straightforward what's going on here. can we extract/name the callback?

Reply:
Maybe it is best to add a comment, one name is not gonna make it more clear I guess.

Reply:
how about both lol

Reply:
add some comment and make the code more readable